components/dashboard/Menu.tsx

```diff
@@ -300,7 +300,7 @@ export const getMenuItems = ({ intl, account, LoggedInUser }): MenuItem[] => {
       section: ALL_SECTIONS.HOST_TAX_FORMS,
       Icon: FileText,
       label: intl.formatMessage({ defaultMessage: 'Tax Forms', id: 'skSw4d' }),
-      if: isHost && Boolean(account.host?.requiredLegalDocuments?.includes('US_TAX_FORM')),
+      if: isFeatureSupported(account, 'TAX_FORMS'),
```

Comment:
This is dependent on if feature returns "UNSUPPORTED" for accounts that should not have this entry, e.g non hosts and non-US hosts.